solr/docker/templates/Dockerfile.body.template

```diff
@@ -45,6 +45,7 @@ ENV SOLR_USER="solr" \
     SOLR_PID_DIR=/var/solr \
     SOLR_LOGS_DIR=/var/solr/logs \
     LOG4J_PROPS=/var/solr/log4j2.xml \
+    SOLR_SERVER_DIR="/opt/solr-${SOLR_VERSION}/server" \
```

Comment:
Would not `/opt/solr/server` suffice?

Reply:
Saw your comment about Jetty canonical path, but I did not understand it :)

Reply:
So I don't fully understand why security manager cares - but the gist is this:

default as configured is `/opt/solr/server` - this gets set as `jetty.home` which gets put in security manager policy to allow/read/write/readlink to `/opt/solr/server`

Jetty when it starts up tries to read from the canonical location where start.jar is located. This is not a symlink but `/opt/solr-10.0.0-SNAPSHOT/server`. I think this is Java saying the current path is `/opt/solr-10.0.0-SNAPSHOT/server` and not `/opt/solr/server`. When Jetty then tries to read `start.ini` (an init file for Jetty) from `/opt/solr-10.0.0-SNAPSHOT/server - the security manager doesn't see that path in the security policy - `/opt/solr/server` is there for `jetty.home`. This causes security manager to deny access and Solr to not start up.

The fix I found is to set `SOLR_SERVER_DIR` to the canonical path (`/opt/solr-10.0.0-SNAPSHOT/server`) which in turn sets `jetty.home` to that path which then updates the security policy to the correct path. Solr install dir and other dirs in the security policy still reference `/opt/solr/...` as needed. This just ends up being a Jetty canonical path thing it looks like.

Reply:
I see. So Java does not understand symlinks.. Could we instead of setting this SOLR_SERVER_DIR, open up `/opt/solr-${SOLR_VERSION}/` in the security manager, so both the canonical and the symlink paths are explicitly allowed? It feels safer.

Reply:
Java understands symlinks - but only if that was the requested path I think. Jetty seems to explicitly request the canonical path.

```
 permission java.io.FilePermission "/opt/solr-*", "read,write,delete,readlink";
  permission java.io.FilePermission "/opt/solr-*${/}-", "read,write,delete,readlink";
```

this just ends up hardcoding a path. Very few paths should be hardcoded in the security policy. Most are passed in as system properties.

another way to do it would be to find a way to expand the symlink in bash and pass it in as another property to the security policy - but I couldn't find a cross platform way to do that.

so setting SOLR_SERVER_DIR seems like the best way forward for now. and it also seems like it is the reason that env variable is exposed...